chrysom/basicClient.go

```diff
@@ -52,18 +55,18 @@ type BasicClientConfig struct {
 	// Bucket partition to be used by this client.
 	Bucket string
 
-	// HTTPClient refers to the client that will be used to send requests.
-	// (Optional) Defaults to http.DefaultClient.
-	HTTPClient *http.Client
-
 	// Auth provides the mechanism to add auth headers to outgoing requests.
 	// (Optional) If not provided, no auth headers are added.
 	Auth Acquirer
+
+	// PullInterval is how often listeners should get updates.
+	// (Optional). Defaults to 5 seconds.
+	PullInterval time.Duration
 }
 
 // BasicClient is the client used to make requests to Argus.
 type BasicClient struct {
-	client       *http.Client
+	client       HTTPClient
```

Comment:
This allows custom http clients since servers like tr1d1um need it. ATM, the current setup forces the user to do some not so great things to accomplish the same thing.